vendor/bat-native-ledger/src/bat/ledger/internal/response/response_sku.cc

```diff
@@ -0,0 +1,276 @@
+//   "currency": "BAT",
+//   "kind": "uphold",
+//   "amount": "1"
+// }
+
+ledger::Result CheckSendExternalTransaction(
+    const ledger::UrlResponse& response) {
+  // Bad Request (400)
+  if (response.status_code == net::HTTP_BAD_REQUEST) {
+    BLOG(0, "Invalid request");
+    return ledger::Result::LEDGER_ERROR;
+  }
+
+  // Not Found (404)
+  if (response.status_code == net::HTTP_NOT_FOUND) {
+    BLOG(0, "Unrecognized transaction suffix");
+    return ledger::Result::NOT_FOUND;
+  }
+
+  // Conflict (409)
+  if (response.status_code == net::HTTP_CONFLICT) {
+    BLOG(0, "External transaction id already submitted");
+    return ledger::Result::LEDGER_ERROR;
+  }
+
+  // Internal Server Error (500)
+  if (response.status_code == net::HTTP_INTERNAL_SERVER_ERROR) {
+    BLOG(0, "Internal server error");
+    return ledger::Result::LEDGER_ERROR;
+  }
+
+  // Created (201)
+  if (response.status_code != net::HTTP_CREATED) {
+    return ledger::Result::LEDGER_ERROR;
+  }
+
+  return ledger::Result::LEDGER_OK;
+}
+
+// Request Url:
+// POST /v1/orders
+//
+// Success:
+// Created (201)
+//
+// Response Format:
+// {
+//   "id": "f2e6494e-fb21-44d1-90e9-b5408799acd8",
+//   "createdAt": "2020-06-10T18:58:21.378752Z",
+//   "currency": "BAT",
+//   "updatedAt": "2020-06-10T18:58:21.378752Z",
+//   "totalPrice": "1",
+//   "location": "brave.com",
+//   "status": "pending",
+//   "items": [
+//     {
+//       "id": "9c9aed7f-b349-452e-80a8-95faf2b1600d",
+//       "orderId": "f2e6494e-fb21-44d1-90e9-b5408799acd8",
+//       "sku": "user-wallet-vote",
+//       "createdAt": "2020-06-10T18:58:21.378752Z",
+//       "updatedAt": "2020-06-10T18:58:21.378752Z",
+//       "currency": "BAT",
+//       "quantity": 4,
+//       "price": "0.25",
+//       "subtotal": "1",
+//       "location": "brave.com",
+//       "description": ""
+//     }
+//   ]
+// }
+
+ledger::SKUOrderPtr ParseOrderCreate(
+    const ledger::UrlResponse& response,
+    const std::vector<ledger::SKUOrderItem>& order_items) {
+  // Bad Request (400)
+  if (response.status_code == net::HTTP_BAD_REQUEST) {
+    BLOG(0, "Invalid request");
+    return nullptr;
+  }
+
+  // Internal Server Error (500)
+  if (response.status_code == net::HTTP_INTERNAL_SERVER_ERROR) {
+    BLOG(0, "Internal server error");
+    return nullptr;
+  }
+
+  // Created (201)
+  if (response.status_code != net::HTTP_CREATED) {
+    return nullptr;
+  }
+
+  base::Optional<base::Value> dictionary =
+      base::JSONReader::Read(response.body);
+  if (!dictionary || !dictionary->is_dict()) {
+    BLOG(0, "Invalid JSON");
+    return nullptr;
+  }
+
+  auto order = ledger::SKUOrder::New();
+
+  const auto* id = dictionary->FindStringKey("id");
+  if (id) {
+    order->order_id = *id;
+  }
+
+  if (order->order_id.empty()) {
+    return nullptr;
+  }
+
+  const auto* total_amount = dictionary->FindStringKey("totalPrice");
+  if (total_amount) {
+    base::StringToDouble(*total_amount, &order->total_amount);
```

Comment:
we need to handle return, see example above

Reply:
Fixed.